src/components/ReportActionItem/IOUAction.js

```diff
@@ -43,29 +63,60 @@ const IOUAction = (props) => {
     const launchDetailsModal = () => {
         Navigation.navigate(ROUTES.getIouDetailsRoute(props.chatReportID, props.action.originalMessage.IOUReportID));
     };
+
+    const shouldShowIOUPreview = (
+        props.isMostRecentIOUReportAction && Boolean(props.action.originalMessage.IOUReportID)
+    ) || props.action.originalMessage.type === CONST.IOU.REPORT_ACTION_TYPE.PAY;
+
+    let shouldShowPendingConversionMessage = false;
+    if (
+        props.iouReport
+        && props.chatReport.hasOutstandingIOU
+        && props.isMostRecentIOUReportAction
+        && props.action.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD
+        && props.network.isOffline
+    ) {
+        shouldShowPendingConversionMessage = IOUUtils.isIOUReportPendingCurrencyConversion(props.reportActions, props.iouReport);
+    }
+
+    let isConverting = false;
+    if (
+        props.iouReport
+        && props.chatReport.hasOutstandingIOU
+        && props.isMostRecentIOUReportAction
+        && !props.network.isOffline
+    ) {
+        const hasPendingRequestsInDifferentCurrency = _.chain(
+            IOUUtils.getIOUReportActions(props.reportActions, props.iouReport, '', '', true),
+        ).some(action => action.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD).value();
+
+        isConverting = !props.network.isOffline && hasPendingRequestsInDifferentCurrency;
```

Comment:
@vitHoracek what do you think about this? Here's what it does:

https://user-images.githubusercontent.com/9680864/205949410-4acea658-4c1e-438e-bc23-d83b90755ecc.mov

The video is not demonstrating the need for it well, but in some cases the report's total keeps changing as we keep receiving data from the backend until it settles down.